### PR TITLE
[WFLY-6501] EJB with AroundConstruct interceptor with Object return t…

### PR DESCRIPTION
**Pull request by treblereel**
…ype fails to deploy
https://issues.jboss.org/browse/WFLY-6501

### CODE REVIEW

**Comment by treblereel** [comment]
windows tests crashed, as usual :-)

**Comment by treblereel** [comment]
retest this please

**Comment by treblereel** [comment]
@bstansberry Could you please take a look at this ?

File: ee/src/main/java/org/jboss/as/ee/component/deployers/LifecycleAnnotationParsingProcessor.java [hunk 1]
```diff
@@ -107,6 +107,9 @@ private void processLifeCycle(final EEModuleDescription eeModuleDescription, fin
             builder.setPostConstruct(methodIdentifier);
         } else if (annotationType == PRE_DESTROY_ANNOTATION) {
             builder.setPreDestroy(methodIdentifier);
+        } else if(annotationType == AROUND_CONSTRUCT_ANNOTATION){
+            String returnType = methodInfo.returnType().name().toString();
+            builder.setAroundConstruct(MethodIdentifier.getIdentifier(returnType, methodInfo.name(),InvocationContext.class.getName()));
         } else {
             builder.setAroundConstruct(methodIdentifier);
         }
```
Comment (bstansberry) [inline ee/src/main/java/org/jboss/as/ee/component/deployers/LifecycleAnnotationParsingProcessor.java]: This 'else' block is now unused.

I think the fix should be done earlier in the method when the object referenced by "methodIdentifier" is created. At that point you have the MethodInfo, can determine the return type, and can precisely create the correct MethodIdentifier. If non-void return types are invalid for POST_CONSTRUCT_ANNOTATION, PRE_DESTROY_ANNOTATION you can also check for and reject that, and if they are not invalid then this class is incorrect for those as well.

**Comment by treblereel** [comment]
retest this please

**Comment by treblereel** [comment]
@bstansberry much better,  could you please take a look at this once again ?

**Comment by bstansberry** [comment]
@n1hility I didn't put a 10.x or 11.x label on this as I don't have a strong opinion on whether it should be in 10.1.

**Pull request merged**